src/python/pants/option/option_types.py

```diff
@@ -802,6 +803,7 @@ def __new__(
                     """
                 )
             ),
+            default=default,
```

Comment:
Lol what. The new error message makes no sense to me. The types are the same according to the error. I assumed `_ListMemberT` was the same name, but not the same value -- but that doesn't seem to be it.

```
src/python/pants/option/option_types.py: note: In member "__new__" of class "ArgsListOption":
[276](https://github.com/pantsbuild/pants/runs/7458162823?check_suite_focus=true#step:9:277)
src/python/pants/option/option_types.py:806:21: error: Argument "default" to
[277](https://github.com/pantsbuild/pants/runs/7458162823?check_suite_focus=true#step:9:278)
"__new__" of "_ListOptionBase" has incompatible type
[278](https://github.com/pantsbuild/pants/runs/7458162823?check_suite_focus=true#step:9:279)
"Union[Callable[[Any], Any], List[_ListMemberT]]"; expected
[279](https://github.com/pantsbuild/pants/runs/7458162823?check_suite_focus=true#step:9:280)
"Union[Callable[[Any], Any], List[_ListMemberT]]"  [arg-type]
[280](https://github.com/pantsbuild/pants/runs/7458162823?check_suite_focus=true#step:9:281)
                default=default,
[281](https://github.com/pantsbuild/pants/runs/7458162823?check_suite_focus=true#step:9:282)
                        ^
```

At what point should we ignore the error?

Reply:
1. File an issue to mypy? 
2. We can type ignore and reference it here

Hopefully callsites are still type checked correctly

Reply:
I've gone with the type ignore for now. @Eric-Arellano or @thejcannon do you have more context for the mypy bug and want to file it? I can do it, but I'd need to spend some time figuring out what is going on here to file a useful bug.

Reply:
Weird right? This goes beyond my understanding of type annotations. I think a type ignore annotation is the way to go.

Reply:
I would expect `mypy` devs to want us to use `main` to reproduce, and bumping mypy in the repo isn't fun. Let's just leave it for now and really hope that its fixed in future versions and we get that "unnecessary type ignore" warning.